File: app/src/main/java/fudi/freddy/mokitosample/share/SharePresent.java
package fudi.freddy.mokitosample.share;

import android.support.annotation.VisibleForTesting;

/**
 * @version 1.0
 * @autor Romell Domínguez
 * @date 8/13/17
 */

public class SharePresent implements ShareContract.ShareUserActions {

    private ShareContract.ShareView shareView;
    private Repository repository = RepositoryShare.getInstance();


    public SharePresent(ShareContract.ShareView shareView) {
        this.shareView = shareView;
    }

    @VisibleForTesting
    void setShareView(ShareContract.ShareView shareView) {
        this.shareView = shareView;
    }

    @Override
    public void saveInput(String string) {
        repository.save(string);
        shareView.clearText();
        shareView.reLoadList();
    }

    @Override
    public void loadInput() {
        String result = repository.load();
        shareView.loadText(result);
    }

    @Override
    public void display(String dat) {
        shareView.loadText(dat);
    }
}

File: app-kotlin/src/main/java/fudi/freddy/mokitosample/share/SharePresent.kt
package fudi.freddy.mokitosample.share

import fudi.freddy.mokitosample.share.ShareContract.ShareUserActions
import fudi.freddy.mokitosample.share.ShareContract.ShareView

/**
 * @version 1.0
 * @autor Romell Domínguez
 * @date 7/25/20
 */
class SharePresent(private var shareView: ShareView) : ShareUserActions {

    private var repository: Repository = RepositoryShare.instance

    internal fun visibleForTesting(repository: Repository) {
        this.repository = repository
    }

    override fun saveInput(string: String?) {
        repository.save(string)
        shareView.clearText()
        shareView.reLoadList()
    }

    override fun loadInput() {
        val result = repository.load()
        shareView.loadText(result)
    }

    override fun display(dat: String?) {
        shareView.loadText(dat)
    }

}